src/simple_server/core.clj

```diff
@@ -16,31 +16,76 @@
 
 (defn new-game-handler []
   (when (game/new-game!)
-    (response "OK- start guessing at /guess")))
+    (response "Ok! start guessing at /new-game/guess")))
 
 (defn guess-handler [guess]
   (condp = (game/guess-answer guess)
-    nil       (-> (response  "You need to supply a guess with /guess?guess=N")
-                  (status 400))
-    :game-over (response  "Congratulations! You win!")
+    nil        (-> (response  "You need to supply a guess with /guess?guess=N")
+                   (status 400))
+    :correct   (response "Congratulations! You win!")
+    :game-over (response "Game over! You lose")
     :too-low   (response "Too low.")
-    :too-high  (response  "Too high.")))
+    :too-high  (response "Too high.")))
+
+(defn authorization-handler
+  ([id]
+   (condp = (game/authorize-user id)
+     nil          (-> (response "Please provide your name to enter the game")
+                      (status 401))
+     :user-exists (response "Username already exists! Please choose another name")
+     :authorized  (response "Ok! Start guessing at /auth-game/<yourname>/guess?guess=<number>")))
+  
+  ;; handle a guess from an authorized player
+  ([id guess]
+   (condp = (game/user-guess id guess)
+     nil           (-> (response  "You need to provide your name and a guess at /auth-game/<yourname>/guess?guess=<yourguess>")
+                       (status 400))
+     :unauthorized (-> (response  "Unauthorized! Please register your name to start the game at /auth-game/user/<yourname>")
+                       (status 401))
+     :correct      (response "Congratulations! You win!")
+     :game-over    (response "Game over! You lose!")
+     :too-low      (response "Too low!")
+     :too-high     (response "Too high!"))))
 
 (defroutes game-routes
-  (GET "/new-game" []                 (new-game-handler))
-  (GET "/guess"    [guess :<< as-int] (guess-handler guess))
-  (ANY "*"         []                 (not-found "Sorry, No such URI on this server!")))
+  #_
+  (GET "/new-game"       
+       []
+       (new-game-handler))
+  #_
+  (GET "/new-game/guess" 
+       [guess :<< as-int] 
+       (guess-handler guess))
+  
+  (GET "/"
+       []
+       (response "Please register your name to start the game at /auth-game/user/<yourname>"))
+  (GET "/auth-game/user/:id" 
+       [id] 
+       (authorization-handler id))
+  (GET "/auth-game/:id/guess" 
```

Comment:
Your solution works, and using URL rewriting is a fine idea.  A few comments:

The main disadvantage of your solution is that you've now complected the "playing the game part" with the "authentication" part; i.e. the function authorization-handler is now quite complex.  Can you think of a way to separate these concerns?

Do you think it makes sense for the simple-server.simple-game namespace to be responsible for authentication?  Why, or why not?